istioctl/cmd/analyze.go

```diff
@@ -163,15 +164,24 @@ istioctl experimental analyze -k -d false
 				fmt.Fprintln(cmd.ErrOrStderr())
 			}
 
-			if len(result.Messages) == 0 {
+			// Filter output by specified level
+			var messages diag.Messages
```

Comment:
I think "outputMessages" or something would be better. Since we're decoupling filtering for output and error I think we should be explicit about having two different buckets.